java/mykola-melnyk/src/com/zoolatech/lecture2/tasks/_2/SimpleRobot.java

```diff
@@ -0,0 +1,226 @@
+            compass = "North";
+        } else if (dir == Direction.SOUTH) {
+            compass = "South";
+        }
+    }
+
+    public void move(int steps) {
+
+        System.out.println("Moving " + steps + " step(s) to " + compass + ".");
+        if (steps > 0) {
+
+            if (dir == Direction.EAST) {
+                if ((x + steps) < W) {
+                    x += steps;
+                } else {
+                    x = W - 1;
+                    System.out.println("Reached Eastern wall! Change direction");
+                }
+            } else if (dir == Direction.WEST) {
+                if ((x - steps) >= 0) {
+                    x -= steps;
+                } else {
+                    x = 0;
+                    System.out.println("Reached Western wall! Change direction");
+                }
+            } else if (dir == Direction.SOUTH) {
+                if ((y + steps) < H) {
+                    y += steps;
+                } else {
+                    y = H - 1;
+                    System.out.println("Reached Southern wall! Change direction");
+                }
+            } else if (dir == Direction.NORTH) {
+                if ((y - steps) >= 0) {
+                    y -= steps;
+                } else {
+                    y = 0;
+                    System.out.println("Reached Northern wall! Change direction");
+                }
+            }
+        } else {
+            System.out.println("Robot made no moves!");
+        }
+    }
+
+    public void moveAndShow(int steps) {
+
+        System.out.println("Moving " + steps + " step(s) to " + compass + ".");
+        if (steps > 0) {
+
+            if (dir == Direction.EAST) {
+                if ((x + steps) < W) {
+                    x += steps;
+                } else {
+                    x = W - 1;
+                    System.out.println("Reached Eastern wall! Change direction");
+                }
+            } else if (dir == Direction.WEST) {
+                if ((x - steps) >= 0) {
+                    x -= steps;
+                } else {
+                    x = 0;
+                    System.out.println("Reached Western wall! Change direction");
+                }
+            } else if (dir == Direction.SOUTH) {
+                if ((y + steps) < H) {
+                    y += steps;
+                } else {
+                    y = H - 1;
+                    System.out.println("Reached Southern wall! Change direction");
+                }
+            } else if (dir == Direction.NORTH) {
+                if ((y - steps) >= 0) {
+                    y -= steps;
+                } else {
+                    y = 0;
+                    System.out.println("Reached Northern wall! Change direction");
+                }
+            }
+            showPosition();
+        } else {
+            System.out.println("Robot made no moves!");
+        }
+    }
+
+    public void turnLeft() {
+        if (dir == Direction.SOUTH) {
```

Comment:
it might be more optimal to use the switch expression here

Reply:
Resolved: rewrote the program code completely.